sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -760,13 +760,13 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
   }
 
   /**
-   * Caches the specified table or view in-memory.
+   * Persist the specified table or view with the default storage level,
```

Comment:
It seems `Caches` better than `Persist`.

Reply:
The default storage level is memory and disk. This docs follows the `dataset.cache`

Reply:
Obviously, it's not persistent If user specify the memory_only and so on.

Reply:
I mean this is following the `dataset.cache`
https://github.com/apache/spark/blob/2e945d7a0692a1d6911d3f9ec74875092fb2cbc0/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L3806-L3812

Reply:
The origin behavior of `persist` is surely persistent.
This PR changes the behavior now. It seems a break change.
I think we should avoid to change `persist`.

Reply:
This pr just pulls out the default storage level to config, so the default bebavior is same with previous.

Reply:
Before this change, `persist` surely write data to disk(means persistent) no matter how many times the user calls.

After this change, `persist` may not write data to disk if some user change the default storage level with `only_memory`.

So I think this change make a break change. It seems not change the logic is better.

Reply:
> Before this change, persist surely write data to disk(means persistent) no matter how many times the user calls.

not exactly, the default storage level of persist is memory and disk, so it's possible only cache it in memory. 

> After this change, persist may not write data to disk if some user change the default storage level with only_memory.

Do you mean user change the default storage level config but actually he does not want to change the default storage level. So why does he change the config ? I'm not sure what are you worried about.

Reply:
Because spark session could be shared with users.

Reply:
I do not think Spark should care about it. Spark provide multi session to do configuration isolation. If all user use one session, then all the Spark configs can not be changed. e.g., AQE configs, ANSI configs, etc. Say, a user want ansi mode but b user do not want ? What's the difference between this storage level config and other sql configs ?

Reply:
> not exactly, the default storage level of persist is memory and disk, so it's possible only cache it in memory.

Uh, The persist uses memory and disk. The method name `persist` is a bit confused. If so, `persist` is not surely persistent. The `cache` is the same as `persist`. The behavior of these two methods is uncertain.

Please ignore the comments left above.